src/components/editor-page/render-context/editor-to-renderer-communicator-context-provider.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { createContext, useContext, useMemo } from 'react'
+import { EditorToRendererCommunicator } from '../../render-page/window-post-message-communicator/editor-to-renderer-communicator'
+
+const EditorToRendererCommunicatorContext = createContext<EditorToRendererCommunicator | undefined>(undefined)
+
+/**
+ * Provides the {@link EditorToRendererCommunicator editor to renderer iframe communicator} that is set by a {@link EditorToRendererCommunicatorContextProvider context provider}.
+ *
+ * @return the received communicator
+ * @throws Error if no communicator was received
+ */
+export const useEditorToRendererCommunicator: () => EditorToRendererCommunicator = () => {
+  const communicatorFromContext = useContext(EditorToRendererCommunicatorContext)
+  if (!communicatorFromContext) {
+    throw new Error('No editor-to-renderer-iframe-communicator received. Did you forget to use the provider component?')
+  }
+  return communicatorFromContext
+}
+
+export const EditorToRendererCommunicatorContextProvider: React.FC = ({ children }) => {
```

Comment:
Missing ESDoc (does not need @return, but should be present)

Reply:
done